src/psij/executors/batch/batch_scheduler_executor.py

```diff
@@ -0,0 +1,539 @@
+        while True:
+            self._poll()
+            time.sleep(self.config.queue_polling_interval)
+
+    def _poll(self) -> None:
+        with self._jobs_lock:
+            if len(self._jobs) == 0:
+                return
+        logger.info('Polling for %s jobs', len(self._jobs))
+        try:
+            out = self.executor._run_command(self.executor.get_status_command(self._jobs.values()))
+            logger.debug('Output from qstat: %s', out)
+            self._poll_error_count = 0
+
+            status_map = self.executor.parse_status_output(out)
+
+            with self._jobs_lock:
+                jobs_copy = dict(self._jobs)
+
+            for native_id, job in jobs_copy.items():
+                try:
+                    status = self._get_job_status(native_id, status_map)
+                except Exception:
+                    status = JobStatus(JobState.FAILED,
+                                       message='Failed to update job status: %s' %
+                                               traceback.format_exc())
+                self.executor._set_job_status(job, status)
+                if status.state.final:
+                    with self._jobs_lock:
+                        del self._jobs[native_id]
+        except subprocess.CalledProcessError as ex:
+            self._handle_poll_error(False, ex, 'Failed to poll for job status. '
+                                               'Error from status command: {}'.format(ex.output))
+        except Exception as ex:
+            msg = traceback.format_exc()
+            self._handle_poll_error(True, ex, 'Failed to poll for job status. '
+                                              'Internal error: {}'.format(msg))
+
+    def _get_job_status(self, native_id: str, status_map: Dict[str, JobStatus]) -> JobStatus:
+        if native_id in status_map:
+            return status_map[native_id]
+        else:
+            return JobStatus(JobState.COMPLETED)
+
+    def _handle_poll_error(self, immediate: bool, ex: Exception, msg: str) -> None:
+        logger.warning('Polling error: %s', msg)
+        self._poll_error_count += 1
+        if immediate or (self._poll_error_count > self.config.queue_polling_error_threshold):
+            self._poll_error_count = 0
+            # pull one job
+            with self._jobs_lock:
+                # We should only poll if there is at least one job, so we should not be in a
+                # situation when we polled and there were no jobs to poll for
+                # Internal errors are a bit different, since they could, in principle, occur
+                # after the last job was processed and removed from self._jobs; in practice,
+                # the code in _poll has the job removal from _jobs as the last possible step
+                assert len(self._jobs) > 0
+                _, job = self._jobs.popitem()
+            status = JobStatus(JobState.FAILED, message=msg)
+            self.executor._set_job_status(job, status)
```

Comment:
The point was to propagate the error to the user in a way that isn't overly dramatic. But then a failure to properly qstat three times in a row does seem fairly dramatic, so it may justify failing all jobs.

Reply:
Fixed. All jobs are being failed now.